packages/embark/src/lib/core/events.js

```diff
@@ -45,22 +54,64 @@ EventEmitter.prototype.request = function() {
 
   log("requesting: ", requestName);
   warnIfLegacy(requestName);
-  return this.emit('request:' + requestName, ...other_args);
+  const listenerName = 'request:' + requestName;
+
+  // if we don't have a command handler set for this event yet,
+  // store it and fire it once a command handler is set
+  if (!this.listeners(listenerName).length) {
+    if(!toFire[listenerName]) {
+      toFire[listenerName] = [];
+    }
+    toFire[listenerName].push(other_args);
+    return;
+  }
+
+  return this.emit(listenerName, ...other_args);
 };
 
 EventEmitter.prototype.setCommandHandler = function(requestName, cb) {
   log("setting command handler for: " + requestName);
   let listener = function(_cb) {
     cb.call(this, ...arguments);
   };
+  const listenerName = 'request:' + requestName;
+
   // unlike events, commands can only have 1 handler
-  this.removeAllListeners('request:' + requestName);
-  return this.on('request:' + requestName, listener);
+  _removeAllListeners.call(this, listenerName);
+
+  // if this event was requested prior to the command handler
+  // being set up, call the callback. Do not remove this listener
+  // so any 
+  const prematureListenerArgs = cloneDeep(toFire[listenerName]);
+  if (prematureListenerArgs) {
```

Comment:
That's an array. Need to check with `prematureListenerArgs.length`

Reply:
Great catch, changed this to an array last minute, but forgot to update the check.

Reply:
Turns out this is not needed as `cloneDeep` returns `undefined` if the item in the array is undefined.

Reply:
Nothing that needs to be done now, but maybe for the future it'll be a tiny bit more robust but always ensuring the type you're expecting:

```
cloneDeep(toFire[listenerName]) || []
```

Obviously JS is very forgiving, but I think we'll be better off in the long run if we rely less on `undefined`. 

Anyways, moving on :)